exporter/datadogexporter/factory.go

```diff
@@ -103,7 +103,8 @@ type factory struct {
 	attributesTranslator     *attributes.Translator
 	attributesErr            error
 
-	wg sync.WaitGroup // waits for agent to exit
+	tracesWG  sync.WaitGroup // waits for agent to exit
+	metricsWG sync.WaitGroup // waits for consumeStatsPayload to exit
```

Comment:
It does not seem like these need to be as fields of factory, if we refactor things a bit they could be inside the metrics and traces methods instead of here (which would avoid accidental `Wait`s on them)